pkg/controller/pravega/bookie.go

```diff
@@ -77,22 +80,27 @@ func MakeBookieStatefulSet(pravegaCluster *v1alpha1.PravegaCluster) *appsv1.Stat
 	}
 }
 
-func makeBookieStatefulTemplate(pravegaCluster *v1alpha1.PravegaCluster) corev1.PodTemplateSpec {
+func MakeBookiePodTemplate(p *v1alpha1.PravegaCluster) corev1.PodTemplateSpec {
 	return corev1.PodTemplateSpec{
 		ObjectMeta: metav1.ObjectMeta{
-			Labels: util.LabelsForBookie(pravegaCluster),
+			Labels:      util.LabelsForBookie(p),
+			Annotations: map[string]string{"pravega.version": p.Spec.Version},
 		},
-		Spec: *makeBookiePodSpec(pravegaCluster.Name, pravegaCluster.Spec.Bookkeeper),
+		Spec: *makeBookiePodSpec(p),
 	}
 }
 
-func makeBookiePodSpec(clusterName string, bookkeeperSpec *v1alpha1.BookkeeperSpec) *corev1.PodSpec {
+func makeBookiePodSpec(p *v1alpha1.PravegaCluster) *corev1.PodSpec {
+	terminationGracePeriodSeconds := int64(0)
```

Comment:
What is the reason for setting grace period seconds to be 0? Thanks

Reply:
Same as above, I'll create a document explaining the upgrade process and the reason behind those decisions :)